packages/mask/src/plugins/Pets/SNSAdaptor/Animate.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react'
+import { makeStyles, useStylesExtends } from '@masknet/theme'
+import { useEssay, useDefaultEssay, useCurrentVisitingUser } from '../hooks'
+import { ModelNFT } from './ModelNFT'
+import { NormalNFT } from './NormalNFT'
+import { ImageType } from '../types'
+import { PluginPetMessages } from '../messages'
+
+const useStyles = makeStyles()(() => ({
+    root: {
+        position: 'fixed',
+        top: 0,
+        left: 0,
+    },
+}))
+
+const AnimatePic = () => {
+    const classes = useStylesExtends(useStyles(), {})
+
+    const [start, setStart] = useState(true)
+    const [refresh, setRefresh] = useState(0)
+
+    const visitor = useCurrentVisitingUser(refresh)
+    const visitorMeta = useEssay(visitor, refresh)
+    const defMeta = useDefaultEssay(visitor)
+    const showMeta = visitorMeta ?? defMeta
+
+    const [hidden, setHidden] = useState(false)
+    const [infoShow, setInfoShow] = useState(false)
+
+    const handleClose = () => setHidden(true)
+    const handleMouseEnter = () => setInfoShow(true)
+    const handleMouseLeave = () => setInfoShow(false)
+
+    const refreshHandle = async (data: number) => setRefresh(data)
+    useEffect(() => {
+        PluginPetMessages.events.setResult.on(refreshHandle)
```

Comment:
put it inside refreshHandle, otherwise it will be recreate once re-render.
```suggestion
    useEffect(() => {
        const refreshHandle = (data: number) => setRefresh(data)
        PluginPetMessages.events.setResult.on(refreshHandle)
```

Reply:
>  it will be recreate once re-render.

the cleanup function in of this`useEffect` will release a recreate `refreshHandle` which never been listen.